### PR TITLE
chore: upgrade packages and linter and fix lint errors

### PR DESCRIPTION
**Pull request by bryans99**
packages upgraded
1. looker components
2. looker sdk
3. looker extension sdks
4. eslint

eslint changes
Fixes most eslint errors and warnings although vscode is still reporting some problems.
The two remaining eslint warnings will require refactoring to fix.
There are a number of eslint overrides relating to defaultProperties that should be fixed properly.
Refactored JoinIcon to not use an eval.
Fixed lots of places where array map was used but should have used forEach.
join.ts needs to be revisited. It has some mouse event code that may not be required.
test data files are excluded from eslint - these were causing real performance problems
and most likely would have generated a lot of errors.
HelpPanelData excluded for the same reason.

other
1. had to change the react-query cache key for the model diagrams
   as the list of git branches was causing a double render. Not
   sure why this is happening now. Upgrade to SDK maybe?
2. Fixed MetadataPanel to use the current explore name. For some reason
   it was using the selected field name so a 404 occured.

### CODE REVIEW

File: src/components/DiagramFrame/FramePanels/SettingsPanel/utils.tsx [hunk 1]
```diff
@@ -24,22 +24,25 @@
 
  */
 import React from 'react'
-import {
+import type {
   ILookmlModel,
   ILookmlModelExplore,
   IGitBranch,
 } from '@looker/sdk/lib/4.0/models'
```
Comment (jkaster) [inline src/components/DiagramFrame/FramePanels/SettingsPanel/utils.tsx:31]: Does this still need to be a deep link?

File: src/utils/urls.ts [hunk 2]
```diff
@@ -24,7 +24,7 @@
 
  */
 
-import {
+import type {
   ILookmlModelExplore,
   ILookmlModelExploreField,
 } from '@looker/sdk/lib/4.0/models'
```
Comment (jkaster) [inline src/utils/urls.ts:30]: deep link

File: src/components/DiagramFrame/FramePanels/SettingsPanel/utils.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (bryans99) [reply src/components/DiagramFrame/FramePanels/SettingsPanel/utils.tsx:31]: Probably not. This was auto fixed by eslint. I'd like to use the new ExtensionProvider40 but I'm not sure if that has been published yet. When it has I'll have another run at this and clean up the vscode issues I am seeing.

File: src/components/DiagramFrame/MetadataPanel/MetadataPanel.tsx [hunk 3]
```diff
@@ -41,18 +41,21 @@ import {
...
-import { ILookmlModel, ILookmlModelExplore } from '@looker/sdk'
+import type { ILookmlModel, ILookmlModelExplore } from '@looker/sdk'
 import { Explore, LogoRings } from '@looker/icons'
-import { VpnKey } from '@styled-icons/material/VpnKey'
-import {
+import { VpnKey } from '@styled-icons/material'
+import type {
   ILookmlModelExploreField,
   ILookmlModelExploreJoins,
 } from '@looker/sdk/lib/4.0/models'
```
[the remark this answers is not in the document]
Comment (bryans99) [reply src/components/DiagramFrame/MetadataPanel/MetadataPanel.tsx:50]: See above :D

File: src/components/DiagramFrame/types.ts [hunk 4]
```diff
@@ -24,10 +24,13 @@
...
  */
 
-import { ILookmlModel, ILookmlModelExplore } from '@looker/sdk/lib/4.0/models'
-import { SelectionInfoPacket } from '../interfaces'
-import { DetailedModel } from '../../utils/fetchers'
-import { DiagrammedModel } from '../../utils/LookmlDiagrammer/'
+import type {
+  ILookmlModel,
+  ILookmlModelExplore,
+} from '@looker/sdk/lib/4.0/models'
```
[the remark this answers is not in the document]
Comment (bryans99) [reply src/components/DiagramFrame/types.ts:30]: And see my response :D

**Pull request merged**